packages/amplify-category-geo/src/__tests__/service-walkthroughs/deviceLocationTrackerWalkthrough.test.ts

```diff
@@ -0,0 +1,151 @@
+    mockAmplifyMeta.geo[secondaryDeviceTrackerName] = { ...mockDeviceTrackerParameters, ...secondaryDeviceTrackerResource };
+    mockAmplifyMeta.geo[mockPlaceIndexResource.resourceName] = mockPlaceIndexResource;
+
+    mockContext.amplify.getUserPoolGroupList = jest.fn().mockReturnValue([mockUserPoolGroup]);
+
+    pathManager.getBackendDirPath = jest.fn().mockReturnValue('');
+    // mock reading the role group permissions
+    const mockTrackerRoleAndGroupPermissionsMap: Record<string, $TSObject> = {};
+    // eslint-disable-next-line max-len
+    mockTrackerRoleAndGroupPermissionsMap[mockDeviceTrackerName] = { roleAndGroupPermissionsMap: mockDeviceTrackerParameters.roleAndGroupPermissionsMap };
+    stateManager.getResourceInputsJson = jest.fn().mockReturnValue(mockTrackerRoleAndGroupPermissionsMap);
+    JSONUtilities.writeJson = jest.fn().mockReturnValue('');
+
+    stateManager.getMeta = jest.fn().mockReturnValue(mockAmplifyMeta);
+    printer.warn = jest.fn();
+    printer.error = jest.fn();
+    printer.success = jest.fn();
+    printer.info = jest.fn();
+    prompter.input = jest.fn().mockImplementation((message: string): Promise<any> => {
+      let mockUserInput = 'mock';
+      if (message === 'Provide a name for the device location tracker:') {
+        mockUserInput = mockDeviceTrackerParameters.name;
+      }
+      return Promise.resolve(mockUserInput);
+    });
+    prompter.pick = jest.fn().mockImplementation((message: string): Promise<any> => {
+      let mockUserInput: string | string[] = 'mock';
+      if (message === 'Restrict access by?') {
+        mockUserInput = 'Both';
+      } else if (message === 'Who can access this device tracker?') {
+        mockUserInput = AccessType.AuthorizedAndGuestUsers;
+      } else if (message === 'Select one or more cognito groups to give access:') {
+        mockUserInput = mockDeviceTrackerParameters.groupPermissions;
+      } else if (message === `What kind of access do you want for ${mockUserPoolGroup} users? Select ALL that apply:`) {
+        mockUserInput = mockDeviceTrackerParameters.roleAndGroupPermissionsMap[mockUserPoolGroup];
+      } else if (message === 'Select the device location tracker you want to update') {
+        mockUserInput = mockDeviceTrackerParameters.name;
+      } else if (message === 'Select the device location tracker you want to set as default:') {
+        mockUserInput = secondaryDeviceTrackerName;
+      } else if (message === 'Select the device location tracker you want to remove') {
+        mockUserInput = mockDeviceTrackerName;
+      } else if (message === 'Specify the data provider for device location tracker. This will be only used to calculate billing.') {
+        mockUserInput = mockDeviceTrackerParameters.dataProvider;
+      }
+      return Promise.resolve(mockUserInput);
+    });
+    prompter.yesOrNo = jest.fn().mockReturnValue(mockDeviceTrackerParameters.isDefault);
```

Comment:
This function is `async`, so it would be more accurate to call `.mockResolvedValue` which is syntactic sugar for the `.mockImplementation` you did for `prompter.pick` above.